internal/storage/vmbus/vmbus.go

```diff
@@ -11,8 +11,10 @@ import (
 	"github.com/pkg/errors"
 )
 
-// WaitForVMBusDevicePath waits for the vmbus device to exist at /sys/bus/vmbus/devices/<vmbusGUIDPattern>...
-func WaitForVMBusDevicePath(ctx context.Context, vmbusGUIDPattern string) (string, error) {
+var WaitForVMBusDevicePath = waitForVMBusDevicePath
```

Comment:
I typically try to only 'fake' the part that does a call I cant control. So basically in this case its the `filepath.Glob` method. So I would make a 

`var filepathGlob = filepath.Glob`
and then in `WaitForVMBusDevicePath` call `filepathGlob(pattern)`. In the unit tests you can now control the response of `files` or `err` and you can handle all other cases via cancellation etc. 

Doing it the way you did there actually leaks the ability to change the `WaitForVMBusDevicePath` method for anyone in the binary which seems odd.

Reply:
That makes sense. Just updated with these changes in the pci tests.